### ISSUE TITLE
Error in running with snakemake

### ISSUE THREAD

**Issue by htahmasbi** [pyiron/pyiron_meltingpoint#18]
I tried to calculate the melting point of fccAl using snakemake and I am getting the following error. Could you please help me in this regard? Thanks in advance.

```shell
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	meltingpoint
	1
Select jobs to execute...

[Mon Jun 26 14:51:48 2023]
rule meltingpoint:
    input: input.json
    output: output.json, script.nbconvert.ipynb, plot.nbconvert.ipynb
    jobid: 0

Activating conda environment: /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a

Activating conda environment: /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
Input Notebook:  ./scripts/script.ipynb
Output Notebook: script.nbconvert.ipynb
Executing:   0%|                                                                                                                                                                  | 0/45 [00:00<?, ?cell/s]Executing notebook with kernel: python3
Executing:   2%|███▍                                                                                                                                                      | 1/45 [00:02<01:44,  2.37s/cell]
Traceback (most recent call last):
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/bin/papermill", line 10, in <module>
    sys.exit(papermill())
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/cli.py", line 250, in papermill
    execute_notebook(
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/execute.py", line 122, in execute_notebook
    raise_for_execution_errors(nb, output_path)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/execute.py", line 234, in raise_for_execution_errors
    raise error
papermill.exceptions.PapermillExecutionError: 
---------------------------------------------------------------------------
Exception encountered at "In [1]":
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 11
      9 from ase.data import reference_states, atomic_numbers
     10 from sklearn.neighbors import KernelDensity
---> 11 from pyiron_atomistics import Project

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_atomistics/__init__.py:4
      1 __version__ = "0.1"
      2 __all__ = []
----> 4 from pyiron_atomistics.project import Project
      5 from pyiron_atomistics.toolkit import AtomisticsTools
      6 from pyiron_atomistics.atomistics.structure.atoms import ase_to_pyiron, pyiron_to_ase, Atoms

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_atomistics/project.py:11
      9 from string import punctuation
     10 from shutil import copyfile
---> 11 from pyiron_base import Settings, ProjectHDFio, JobType, JobTypeChoice, Project as ProjectCore, Creator as CreatorCore
     12 try:
     13     from pyiron_base import ProjectGUI

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/__init__.py:5
      1 # coding: utf-8
      2 # Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
      3 # Distributed under the terms of "New BSD License", see the LICENSE file.
----> 5 from pyiron_base.state import state
      6 # API of the pyiron_base module - in alphabetical order
      7 from pyiron_base.generic.factory import PyironFactory

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/state/__init__.py:18
      4 """
      5 The `state` module holds (almost!) all the code for defining the global state of a pyiron instance.
      6 Such "global" behaviour is achieved by using the `Singleton` metaclass to guarantee that each class only even has a
   (...)
     14 compatibility.
     15 """
     17 from pyiron_base.state.logger import logger as _logger
---> 18 from pyiron_base.database.manager import database as _database
     19 from pyiron_base.state.publications import publications as _publications
     20 from pyiron_base.state.queue_adapter import queue_adapters as _queue_adapters

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/database/manager.py:238
    234         self._database_is_disabled = s.configuration["disable_database"]
    235         self.open_connection()
--> 238 database = DatabaseManager()

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/generic/util.py:39, in Singleton.__call__(cls, *args, **kwargs)
     37 def __call__(cls, *args, **kwargs):
     38     if cls not in cls._instances:
---> 39         cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
     40     return cls._instances[cls]

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/database/manager.py:31, in DatabaseManager.__init__(self)
     29 self._use_local_database = False
     30 self._database_is_disabled = s.configuration["disable_database"]
---> 31 self.open_connection()

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/database/manager.py:123, in DatabaseManager.open_connection(self)
    118 """
    119 Internal function to open the connection to the database. Only after this function is called the database is
    120 accessable.
    121 """
    122 if self._database is None and not self.database_is_disabled:
--> 123     self._database = DatabaseAccess(
    124         self.sql_connection_string,
    125         self.sql_table_name,
    126         timeout=self.connection_timeout
    127     )

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/database/generic.py:418, in DatabaseAccess.__init__(self, connection_string, table_name, timeout)
    415     raise ValueError("Connection to database failed: " + str(except_msg))
    417 self._chem_formula_lim_length = 50
--> 418 self.__reload_db()
    419 self.simulation_table = HistoricalTable(str(table_name), self.metadata)
    420 self.metadata.create_all()

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_base/database/generic.py:569, in DatabaseAccess.__reload_db(self)
    562 def __reload_db(self):
    563     """
    564     Reload database
    565 
    566     Returns:
    567 
    568     """
--> 569     self.metadata = MetaData(bind=self._engine)
    570     self.metadata.reflect(self._engine)

TypeError: __init__() got an unexpected keyword argument 'bind'

[Mon Jun 26 14:51:54 2023]
Error in rule meltingpoint:
    jobid: 0
    output: output.json, script.nbconvert.ipynb, plot.nbconvert.ipynb
    conda-env: /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
    shell:
        export PYIRONRESOURCEPATHS='./resources'; export PYIRONPROJECTPATHS='.'; papermill ./scripts/script.ipynb script.nbconvert.ipynb -k 'python3' -p input_file 'input.json' -p output_file 'output.json'; papermill ./scripts/plot.ipynb plot.nbconvert.ipynb -k 'python3' -p input_file 'output.json'
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job meltingpoint since they might be corrupted:
script.nbconvert.ipynb
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /home/hossein/codes/pyiron_meltingpoint/.snakemake/log/2023-06-26T145148.576858.snakemake.log
```

**Comment by pmrv** [comment]
Did you install the latest pyiron or the version specified [here](https://github.com/pyiron/pyiron_meltingpoint/blob/master/envs/melting.yaml)? If the former try making a new env with the version specificed here by running

`conda -n my_env -f envs/melting.yaml`

from the repository root.

**Comment by pmrv** [comment]
Something you might try is checking the version of `sqlalchemy` that is installed.

```bash
conda activate /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
conda list sqlalchemy
```

If it is higher than verions `1.4.48`, try to install that explicitely

```bash
conda activate /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
conda install -c conda-forge 'sqlalchemy==1.4.48'
```

**Comment by htahmasbi** [comment]
> Something you might try is checking the version of `sqlalchemy` that is installed.
> 
> ```shell
> conda activate /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
> conda list sqlalchemy
> ```
> 
> If it is higher than verions `1.4.46`, try to install that explicitely
> 
> ```shell
> conda activate /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
> conda install -c conda-forge 'sqlalchemy==1.4.46'
> ```
Thanks, @pmrv for your response.
I tried this quickly, this was the version of ```sqlalchemy```:
```shell
# Name                    Version                   Build  Channel
sqlalchemy                2.0.16           py39hd1e30aa_0    conda-forge
```
After installing version 1.4.46, again I get another error now after 47% progress in executing: 
```shell
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	meltingpoint
	1

[Tue Jun 27 13:09:07 2023]
rule meltingpoint:
    input: input.json
    output: output.json, script.nbconvert.ipynb, plot.nbconvert.ipynb
    jobid: 0

Activating conda environment: /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
Input Notebook:  ./scripts/script.ipynb
Output Notebook: script.nbconvert.ipynb
Executing:   0%|                                                                                                                                                                                                                                                                      | 0/47 [00:00<?, ?cell/s]Executing notebook with kernel: python3
Executing:  47%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▍                                                                                                                                      | 22/47 [00:05<00:06,  3.76cell/s]
Traceback (most recent call last):
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/bin/papermill", line 10, in <module>
    sys.exit(papermill())
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/cli.py", line 250, in papermill
    execute_notebook(
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/execute.py", line 122, in execute_notebook
    raise_for_execution_errors(nb, output_path)
  File "/home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/papermill/execute.py", line 234, in raise_for_execution_errors
    raise error
papermill.exceptions.PapermillExecutionError: 
---------------------------------------------------------------------------
Exception encountered at "In [21]":
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[21], line 1
----> 1 basis.plot3d()

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_atomistics/atomistics/structure/atoms.py:1144, in Atoms.plot3d(self, mode, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera, opacity)
   1121 def plot3d(
   1122     self,
   1123     mode='NGLview',
   (...)
   1142     opacity=1.0
   1143 ):
-> 1144     return self.visualize.plot3d(
   1145         mode=mode,
   1146         show_cell=show_cell,
   1147         show_axes=show_axes,
   1148         camera=camera,
   1149         spacefill=spacefill,
   1150         particle_size=particle_size,
   1151         select_atoms=select_atoms,
   1152         background=background,
   1153         color_scheme=color_scheme,
   1154         colors=colors,
   1155         scalar_field=scalar_field,
   1156         scalar_start=scalar_start,
   1157         scalar_end=scalar_end,
   1158         scalar_cmap=scalar_cmap,
   1159         vector_field=vector_field,
   1160         vector_color=vector_color,
   1161         magnetic_moments=magnetic_moments,
   1162         view_plane=view_plane,
   1163         distance_from_camera=distance_from_camera,
   1164         opacity=opacity,
   1165     )

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_atomistics/atomistics/structure/_visualize.py:104, in Visualize.plot3d(self, mode, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera, opacity)
     50 """
     51 Plot3d relies on NGLView or plotly to visualize atomic structures. Here, we construct a string in the "protein database"
     52 
   (...)
    101     * The colour interpretation of some hex codes is weird, e.g. 'green'.
    102 """
    103 if mode=='NGLview':
--> 104     return self._plot3d(
    105         show_cell=show_cell,
    106         show_axes=show_axes,
    107         camera=camera,
    108         spacefill=spacefill,
    109         particle_size=particle_size,
    110         select_atoms=select_atoms,
    111         background=background,
    112         color_scheme=color_scheme,
    113         colors=colors,
    114         scalar_field=scalar_field,
    115         scalar_start=scalar_start,
    116         scalar_end=scalar_end,
    117         scalar_cmap=scalar_cmap,
    118         vector_field=vector_field,
    119         vector_color=vector_color,
    120         magnetic_moments=magnetic_moments,
    121         view_plane=view_plane,
    122         distance_from_camera=distance_from_camera,
    123     )
    124 elif mode=='plotly':
    125     return self._plot3d_plotly(
    126         camera=camera,
    127         particle_size=particle_size,
   (...)
    132         opacity=opacity,
    133     )

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/pyiron_atomistics/atomistics/structure/_visualize.py:283, in Visualize._plot3d(self, show_cell, show_axes, camera, spacefill, particle_size, select_atoms, background, color_scheme, colors, scalar_field, scalar_start, scalar_end, scalar_cmap, vector_field, vector_color, magnetic_moments, view_plane, distance_from_camera)
    228 """
    229 Plot3d relies on NGLView to visualize atomic structures. Here, we construct a string in the "protein database"
    230 ("pdb") format, then turn it into an NGLView "structure". PDB is a white-space sensitive format, so the
   (...)
    280     * The colour interpretation of some hex codes is weird, e.g. 'green'.
    281 """
    282 try:  # If the graphical packages are not available, the GUI will not work.
--> 283     import nglview
    284 except ImportError:
    285     raise ImportError(
    286         "The package nglview needs to be installed for the plot3d() function!"
    287     )

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/__init__.py:4
      1 import warnings
      3 # for doc
----> 4 from . import adaptor, datafiles, show, widget
      5 from ._version import get_versions
      6 from .adaptor import *

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/show.py:13
      3 from . import datafiles
      4 from .adaptor import (ASEStructure, ASETrajectory, BiopythonStructure,
      5                       FileStructure, HTMDTrajectory, IODataStructure,
      6                       IOTBXStructure, MDAnalysisTrajectory, MDTrajTrajectory,
   (...)
     11                       RdkitStructure,
     12                       TextStructure)
---> 13 from .widget import NGLWidget
     15 __all__ = [
     16     'demo',
     17     'show_pdbid',
   (...)
     40     'show_biopython',
     41 ]
     44 def show_pdbid(pdbid, **kwargs):

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/widget.py:19
     15 from traitlets import (Bool, CaselessStrEnum, Dict, Instance, Int, Integer,
     16                        List, Unicode, observe, validate)
     17 import traitlets
---> 19 from . import color, interpolate
     20 from .adaptor import Structure, Trajectory
     21 from .component import ComponentViewer

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/color.py:114
    110         else:
    111             raise ValueError(f"{obj} must be either list of list or string")
--> 114 ColormakerRegistry = _ColormakerRegistry()

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/base.py:10, in _singleton.<locals>.getinstance()
      8 def getinstance():
      9     if cls not in instances:
---> 10         instances[cls] = cls()
     11     return instances[cls]

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/color.py:47, in _ColormakerRegistry.__init__(self, *args, **kwargs)
     45 try:
     46     get_ipython() # only display in notebook
---> 47     self._ipython_display_()
     48 except NameError:
     49     pass

File ~/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a/lib/python3.9/site-packages/nglview/color.py:54, in _ColormakerRegistry._ipython_display_(self, **kwargs)
     52 if self._ready:
     53     return
---> 54 super()._ipython_display_(**kwargs)

AttributeError: 'super' object has no attribute '_ipython_display_'

[Tue Jun 27 13:09:15 2023]
Error in rule meltingpoint:
    jobid: 0
    output: output.json, script.nbconvert.ipynb, plot.nbconvert.ipynb
    conda-env: /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/941abb5a
    shell:
        export PYIRONRESOURCEPATHS='./resources'; export PYIRONPROJECTPATHS='.'; papermill ./scripts/script.ipynb script.nbconvert.ipynb -k 'python3' -p input_file 'input.json' -p output_file 'output.json'; papermill ./scripts/plot.ipynb plot.nbconvert.ipynb -k 'python3' -p input_file 'output.json'
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job meltingpoint since they might be corrupted:
script.nbconvert.ipynb
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /home/hossein/codes/pyiron_meltingpoint/.snakemake/log/2023-06-27T130907.221956.snakemake.log
```
I will try your first suggestion.

**Comment by htahmasbi** [comment]
> Did you install the latest pyiron or the version specified [here](https://github.com/pyiron/pyiron_meltingpoint/blob/master/envs/melting.yaml)? If the former try making a new env with the version specificed here by running
> 
> `conda -n my_env -f envs/melting.yaml`
> 
> from the repository root.
> 
> EDIT: Hm, it seems snakemake should be doing the correct thing automatically. @lfzhu-phys @jan-janssen Any idea?

I checked the versions, in my current conda environment the version of all packages is the same as ```melting.yaml```.

**Comment by pmrv** [comment]
Can you try to downgrade `ipywidgets==7` or upgrade `nglview==3.0.6`?  I'm just guessing here, but your error is ngl related and I remember we had problems around those versions.

**Comment by htahmasbi** [comment]
> Can you try to downgrade `ipywidgets==7` or upgrade `nglview==3.0.6`? I'm just guessing here, but your error is ngl related and I remember we had problems around those versions.

I could upgrade `nglview==3.0.6` and again got an error after 57% progress. There is a problem with downgrading `ipywidgets`, conda can not collect the package!

**Comment by htahmasbi** [comment]
> ipywidgets==7.7.5

Finally I could install `ipywidgets==7.7.5` but unfortunately the error persists :(

**Comment by hainm** [comment]
Make sure that you are actually using `ipywidgets==7.7.5` 
(or even better, please upgrade to nglview 3.0.6 that works with everything, including lab or notebook or different ipywidgets versions).

**Comment by htahmasbi** [comment]
> Make sure that you are actually using `ipywidgets==7.7.5` (or even better, please upgrade to nglview 3.0.6 that works with everything, including lab or notebook or different ipywidgets versions).

It does not work.

**Comment by hainm** [comment]
can you print the version inside the notebook that is showing the error with `AttributeError: 'super' object has no attribute '_ipython_display_'`?

**Comment by htahmasbi** [comment]
Which notebook? I am running using the command `snakemake --use-conda --cores 1` .

**Comment by htahmasbi** [comment]
> ```shell
> conda activate /home/hossein/codes/pyiron_meltingpoint/.snakemake/conda/
> ```

Unfortunately, did not help.

**Comment by jan-janssen** [comment]
Ok, can you try to replace the content in the `envs/melting.yaml` with: 
```
channels:
- conda-forge
- bioconda
dependencies:
- python=3.10
- lammps=2023.03.28
- pyiron_atomistics=0.3.0
- pyiron_base=0.6.1
- jupyter_client=8.3.0
- papermill=2.3.4
- scikit-learn=1.2.2
- nglview=3.0.6
- mscorefonts=0.0.1
- numpy=1.24.4
```

**Comment by pmrv** [comment]
Thanks @hainm and @jan-janssen to hop in!

**Comment by jan-janssen** [comment]
I released a new `1.3` version of the melting point simulation protocol, it now supports the latest version of pyiron_atomistics and nglview without the need to modify the environment manually:  https://github.com/pyiron/pyiron_meltingpoint/releases/tag/release-v1.3
